Feat/home scene

Hikaru2238C opened the pull request:
動作
ホームシーンに入るとワードが円形に出てくるのでcとdをドラッグするとFranceができる

uwatcha committed:
Merge branch 'feat/home_scene' of github.com:DiverWorlds/WordWorlds into feat/home_scene

uwatcha committed:
Merge branch 'feat/home_scene' of github.com:DiverWorlds/WordWorlds into feat/home_scene

uwatcha committed:
Merge branch 'develop' of github.com:DiverWorlds/WordWorlds into feat/home_scene

uwatcha committed:
DatabaseのInitializeをAwakeにした。

uwatcha committed:
Merge branch 'feat/home_scene' of github.com:DiverWorlds/WordWorlds into feat/home_scene

Hikaru2238C committed:
ドラッグ&ドロップで世界を作る基礎機構だけ

Hikaru2238C committed:
ワード選択画面のクオリティアップと実装を少し進めました！

Hikaru2238C committed:
ドラッグ&ドロップと合成先の予想までできました

Hikaru2238C committed:
ドラッグアンドドロップで世界を作るのを実装中

uwatcha committed:
Merge branch 'feat/home_scene' of github.com:DiverWorlds/WordWorlds into feat/home_scene

uwatcha committed:
Merge branch 'develop' of github.com:DiverWorlds/WordWorlds into feat/home_scene

uwatcha committed:
git pull origin developした

uwatcha committed:
DevSearchWorldDatabaseをSearchWorldDatabaseに名前を戻した

Hikaru2238C committed:
ワード選択→世界作るの流れが一旦できました

uwatcha committed:
Merge branch 'develop' of github.com:DiverWorlds/WordWorlds into feat/home_scene

TakegawaLab commented on Assets/Images/Textures/3DCameraView.renderTexture:
```diff
@@ -0,0 +1,39 @@
+%YAML 1.1
```
TexturesフォルダはAssets直下においた方がよいかもと思った
理由：ImageとTextureは別物だから．
だけど，使用場面が近い２つだから，今の方がよければそれでも良い
もしくは，見た目フォルダ的なのを作って，その中にImagesとTexturesを配置する？

TakegawaLab commented on Assets/Prefabs/InventoryUI/DraggableWordUI.prefab:
```diff
@@ -0,0 +1,320 @@
+%YAML 1.1
```
Text (TMP)オブジェクト不要かも

TakegawaLab commented on Assets/Prefabs/InventoryUI/WordDropper.prefab:
WordDropperという名前だと，Wordをつかむカーソルのイメージを持った．
改名すべし．
今思いついた案だと，ItemWordSetAreaとか？

TakegawaLab commented on Assets/Prefabs/InventoryUI/WordDropper.prefab:
オブジェクト名改善
countText -> CountText
Word1, 2 -> ItemWorld1, 2

TakegawaLab commented on Assets/Scenes/Develop/Dev_HomeScene/Main Camera Profile.asset:
Scenes/CameraProfileみたいなフォルダを作ってそこに置くといいかも
HomeSceneでしか使わない予定ならHomeSceneCameraProfileみたいな名前にしてみて

TakegawaLab commented on Assets/ScriptableObjects/ItemWord/Test/ItemWordDatabase.asset:
ItemWord内に保存しているので，各ItemWordの名前の先頭にword_等はフヨウラ
そして，
Under->UnderWear
のように，略さない名前にしてみて（ファイル名やログを見たときに何のItemWordか明瞭にするため）

TakegawaLab reviewed (changes requested):
**見つけたバグ**
- 条件は不明だが，ワードを掴もうとすると，反時計回りの方向に１つずれたワードを掴んだことになってしまう
https://github.com/user-attachments/assets/0bc17678-d83c-475a-986b-9efe406ae85d

- 「下着」と「喪失」をドラッグしても何も起きずにキャンセルされた時，ログでは下着を２度選んだことになっている．
<img width="352" alt="image" src="https://github.com/user-attachments/assets/3638750e-46d0-4689-9ca5-80cf9b516889" />

- cとdを選んで生成するか確認された後にキャンセルを押すと，Cubeが表示され続ける．
- 一度しか発生しなかったが，フランスを生成した後にゲームを終了すると，「何かのオブジェクトがDestroyされないまま終了しました」のようなエラーが発生した（エラー文も消えてしまったので，軽く調べてわかんなければスルーでよし）

uwatcha commented:
間違えてTakegawaLabアカウントでレビューしちゃったよ最悪
あと，上記のレビューはScript以外に関する部分です．

Hikaru2238C committed:
プログラムに一部修正とコメントをつけました

uwatcha committed:
Merge branch 'feat/home_scene' of github.com:DiverWorlds/WordWorlds into feat/home_scene

uwatcha committed:
HomeManagerの不要な行削除
HomeSceneの不要なオブジェクト削除
HomeManager配下にHomeSceneでのみ使うオブジェクトを配置

uwatcha committed:
WordDropperをItemWordDropAreaに改名

uwatcha committed:
フォルダInventoryUIをItemWordInventoryに改名
DraggableWordUIをDraggableItemWordに改名
ItemWordDropAreaのフィールドを改名
それに合わせてSceneのオブジェクトも改名

uwatcha committed:
dropCounterTextの更新を関数化

uwatcha committed:
座標の更新を関数化した

uwatcha committed:
DraggableInventoryをItemWordInventoryに改名
DraggableItemWordをDraggableItemWords以下に配置するようにした．
その結果，ドラッグができなくなっている．

uwatcha committed:
ItemWordInventoryUIでのInvデータ使用が，Invの初期化より早くて読み込めなかった問題修正．
DontDestroyのAwakeを順序づけて呼び出せる仕組みを作成．

uwatcha committed:
ItemWordInventoryUIでのInvデータ使用が，Invの初期化より早くて読み込めなかった問題修正．
DontDestroyのAwakeを順序づけて呼び出せる仕組みを作成．

uwatcha committed:
Merge branch 'feat/home_scene' of github.com:DiverWorlds/WordWorlds into feat/home_scene

uwatcha committed:
ItemWordDropAreaを改善し，ドロップしたItemWordも再度ドラッグできるようになった(動作未確認)

uwatcha committed:
DraggableItemWordの当たり判定を縮め，隣のItemWordを掴んでしまうことがないようにした．

uwatcha committed:
OnEndDrag時に，ItemWordがAreaのRectTransform内にあるかどうかで判定するようにした．
それに合わせてSceneの構成を改善．(BackGroundPanelではなく，ItemWordDropAreaが本のImageを持つ)

uwatcha committed:
AwakeControllerが使えなかったので，DefaultExecutionOrderで実行順を制御するようにしたらエラーが消えた

uwatcha committed:
ItemWordInventoryの時点でSaveDataからアイテム一覧が読み込めていない問題を修正

uwatcha committed:
ドラッグ＆ドロップ判定ができた．しかし，ドロップ判定が小さく，ドロップエリア内でドロップするとエラー

uwatcha committed:
DraggableItemWordがItemWordDropAreaの参照がなかった問題を修正

uwatcha committed:
可能な範囲で，終了時にエラーが出ないようにした．

uwatcha committed:
プルリク提出時の挙動フローを再現できた．

uwatcha committed:
HomeSceneを改名し，適切な位置に配置

uwatcha committed:
ItemWordのMaterialを再アタッチ. TMP版は赤くするように戻した．

uwatcha committed:
ドロップ時に大きさを元のように変更するようにした．

uwatcha committed:
ドラッグ&ドロップがArea内外問わず自由にできるようになった．

uwatcha committed:
PredictCanvas起動中はD&Dできなくした

uwatcha committed:
細かいバグ修正

uwatcha committed:
引き続きバグ修正

uwatcha committed:
2つともRecipeにないItemWordでPeekしたときに，「同じItemWordが選択された」というログが出る問題を修正

uwatcha committed:
Recipeにない組み合わせの2つのItemWordがドロップされてそれが元の位置に戻るとき，2つ目の大きさが元に戻らない問題を修正

uwatcha committed:
ItemWordのPosition変更の際に，lastPositionの更新漏れがあった．

uwatcha committed:
Refactor drop point handling in ItemWordDropArea to use RectTransform and initialize dropPoints correctly

uwatcha committed:
ItemWordの位置をDropArea内外で変更しなかった時，ドロップ位置がその場になってしまう問題を修正

uwatcha committed:
DropPointの座標と実際にドロップされる位置のずれを解消

uwatcha committed:
オブジェクトの描画順を，
本 -> DropAreaHint -> DraggableItemWord
から，
本 -> DraggableItemWord -> DropAreaHint
の順に変更

uwatcha committed:
ItemWordInventoryUIが２つ存在していたので片方消す．

uwatcha committed:
不要なコメントやusingを削除

uwatcha committed:
PredictCanvasが出る時，DropAreaHintCanvasが消えるようにした．

uwatcha commented:
途中からHomeSceneの作業権限が僕に移ったため，プルリクを作り直す．
